diff --git a/crates/biome_fmt/pkg/biome_fmt.js b/crates/biome_fmt/pkg/biome_fmt.js
index 9346f58..821d708 100644
--- a/crates/biome_fmt/pkg/biome_fmt.js
+++ b/crates/biome_fmt/pkg/biome_fmt.js
@@ -190,11 +190,11 @@ function debugString(val) {
 }
 /**
 * @param {string} src
-* @param {string} filename
+* @param {string | undefined} [filename]
 * @param {Config | undefined} [config]
 * @returns {string}
 */
-export function format(src, filename, config) {
+export function format(src, filename = "", config) {
     let deferred4_0;
     let deferred4_1;
     try {
